pandas/io/parsers.py

```diff
@@ -1568,12 +1566,8 @@ def ix(col):
                 return col
             raise ValueError('Index %s invalid' % col)
 
-        to_remove = []
-        index = []
-        for idx in self.index_col:
-            i = ix(idx)
-            to_remove.append(i)
-            index.append(data[i])
+        to_remove = [ix(idx) for idx in self.index_col]
```

Comment:
-0 on this change since now `self.index_col` is essentially looped over twice

Reply:
Let's revert this actually.  The for-loop is clearer as to what's going on.

Reply:
Sure, can do.
Just for info though, using list comprehensions seems to be significantly faster (at least in some very basic test cases here on my side, Python3.7)

Reply:
Percentages are a little bit deceiving, since I don't know what your starting benchmark times were.  Do you mind running our benchmarks to demonstrate the difference?

Reply:
Definitely in toy examples I am getting similar results as you @phil65. If the ASV benchmarks shows a performance enhancement overall (io benchmarks in particular), I'd be okay with this change.